### DISCUSSION TITLE
Unable to use this flake on darwin

### DISCUSSION THREAD

**Discussion by ashwinmathi** [srid/haskell-flake#47] [General]
I've tried both duplicating the haskell-template repo and starting with a fresh repo and adding the flake specified in this repo to no avail.

Upon running nix develop, it seems to properly download all the dependencies and stuff so after creating the flake.lock I get this error

`
error: flake 'path:/Users/$USER/Documents/Workspace/$PROJNAME' does not provide attribute 'devShells.x86-64-darwin.default', 'devShell.x86-64-darwin', 'packages.x86-64-darwin.default' or 'defaultPackage.x86-64-darwin'
       Did you mean one of devShells or x86_64-darwin?
`

From what I can tell the default package/devShell isn't getting set on darwin but I don't have any other operating systems to check this with.

The only resource I was able to find that helped me understand this a little more was [this issue ](https://github.com/NixOS/nix/issues/6879) on the o.g. nix repo, but that seems to be pretty specific to the 'nix flake new' command which I'm not using for this. (EDIT: the proposed solution seems to be to use flake-utils whereas this flake uses flake-parts)

Is there something I should manually be changing to let this work on macOS?

Thank you for your time and energy maintaining this package!!

**Comment by srid** [comment]
Can you share your `flake.nix`? Do you have a project named "default"? Did you set "packages.default"? 

What is your Nix version?

**Comment by ashwinmathi** [comment]
Nix Version: 2.11.1

I have an attribute "default" set in the haskellProjects sections if I'm understanding what you mean properly and yeah I have packages.default set here:


```
# flake.nix:
{
  inputs = {
    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
    flake-parts.url = "github:hercules-ci/flake-parts";
    haskell-flake.url = "github:srid/haskell-flake";
  };
  outputs = inputs@{ self, nixpkgs, flake-parts, ... }:
    flake-parts.lib.mkFlake { inherit self; } {
      systems = nixpkgs.lib.systems.flakeExposed;
      imports = [ inputs.haskell-flake.flakeModule ];
      perSystem = { self', pkgs, ... }: {
        haskellProjects.default = {
          packages = { 
            # You can add more than one local package here.
            basic-package.root = ./.;  # Assumes ./basic-package.cabal
          };
          # buildTools = hp: { fourmolu = hp.fourmolu; ghcid = null; };
          # overrides = self: super: { };
          # hlintCheck.enable = true;
          # hlsCheck.enable = true;
        };
        # haskell-flake doesn't set the default package, but you can do it here.
        packages.default = self'.packages.basic-package;
      };
    };
}
```

**Reply by srid** [reply]
That looks right to me.

I have an M1 mac, and with Nix 2.11.0 https://github.com/srid/haskell-template works fine here.

Something about the string `'path:/Users/$USER/Documents/Workspace/$PROJNAME'` in your error message looks odd. Anyway, I'd suggest two things:

- Create a github repo containing the minimal project whose `nix develop` errors out.
- Ask in Nix bug tracker or chat forum, as so far I don't see a reason to believe that this is unique to haskell-flake.

**Reply by ashwinmathi** [reply]
Gotcha, thanks so much for the help. Just to clarify I edited the original error message to make it more general so i took out my username and the projectname.

**Reply by ashwinmathi** [reply]
https://discourse.nixos.org/t/cant-use-nix-develop-on-x86-64-darwin/23205

Problem solved - just user error

**Comment by roberth** [comment]
`nix flake show` might reveal something, but first `nix flake lock --update-input flake-parts`, as I've just fixed `nix flake show` support.

**Reply by srid** [reply]
Be sure to make the systems list a singleton before doing this; eg.: `systems = [ "x86_64-darwin" ];`.

https://nixos.wiki/wiki/Haskell#IFD_and_Haskell
